### PR TITLE
shims/super/cc: fix most style errors

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?
- [ ] Have you successfully run `brew man` locally and committed any changes?

-----

`brew style` "skips" over `shims/super/cc` as it seems to see it as a shell script. I temporarily deleted the [`#!/bin/sh` section](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/shims/super/cc#L1-L8) and then ran `brew style --fix`. Since I'm not sure if `shims/super/cc` supports the same Ruby syntax as the rest of the repository, I'm marking this PR as a draft

Here are the remaning style errors: (The line numbers are all off by 8 since I deleted `#!/bin/sh`)

```
Library/Homebrew/shims/super/cc:1:1: C: [Correctable] Missing frozen string literal comment.
#!/usr/bin/env ruby -W0
^
Library/Homebrew/shims/super/cc:18:1: C: Missing top-level class documentation comment.
class Cmd
^^^^^
Library/Homebrew/shims/super/cc:122:5: W: Duplicate branch body detected.
    when :cpp ...
    ^^^^^^^^^
Library/Homebrew/shims/super/cc:223:5: W: Duplicate branch body detected.
    else
    ^^^^
Library/Homebrew/shims/super/cc:239:5: W: Duplicate branch body detected.
    elsif path.start_with?(prefix, cachedir, tmpdir) ...
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Library/Homebrew/shims/super/cc:244:5: W: Duplicate branch body detected.
    elsif prefix != "/usr/local" && path.start_with?("/usr/local") ...
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Library/Homebrew/shims/super/cc:249:5: W: Duplicate branch body detected.
    elsif mac? ...
    ^^^^^^^^^^
Library/Homebrew/shims/super/cc:251:5: W: Duplicate branch body detected.
    else
    ^^^^
```

### CODE REVIEW

File: Library/Homebrew/shims/super/cc [hunk 1]
```diff
@@ -435,15 +436,15 @@ if __FILE__ == $PROGRAM_NAME
 
   ####################################################################### main
 
-  dirname, basename = File.split($0)
+  dirname, basename = File.split($PROGRAM_NAME)
```
Comment (reitermarkus) [inline Library/Homebrew/shims/super/cc:439]: Might need `require "English"` for this.

**Comment by MikeMcQuaid** [comment]
> `brew style` "skips" over `shims/super/cc` as it seems to see it as a shell script. I temporarily deleted the [`#!/bin/sh` section](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/shims/super/cc#L1-L8) and then ran `brew style --fix`. Since I'm not sure if `shims/super/cc` supports the same Ruby syntax as the rest of the repository, I'm marking this PR as a draft

Is there any way to have these checked explicitly somehow? Maybe a `.rb` symlink somewhere we can tell `rubocop` to follow?

File: Library/Homebrew/shims/super/cc (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/shims/super/cc:439]: Yup 👍🏻

File: Library/Homebrew/shims/super/cc [hunk 2]
```diff
@@ -51,7 +51,7 @@ class Cmd
     elsif ["ld", "ld.gold", "gold"].include? @arg0
       :ld
     elsif @args.include? "-c"
-      if @arg0 =~ /(?:c|g|clang)\+\+/
+      if /(?:c|g|clang)\+\+/.match?(@arg0)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/shims/super/cc:54]: Maybe this regex could/should be a constant

**Review by MikeMcQuaid — COMMENTED** [review]
Nice work here! Would be nice if we can figure out a way to avoid these style changes backsliding.

**Comment by SeekingMeaning** [comment]
> Is there any way to have these checked explicitly somehow? Maybe a `.rb` symlink somewhere we can tell `rubocop` to follow?

I tried a `.rb` symlink but `rubocop` trips over the `#!/bin/sh` section; I'll need to look into this further

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
> I tried a `.rb` symlink but `rubocop` trips over the `#!/bin/sh` section; I'll need to look into this further

Given this I'm fine with this just being merged as-is 👍🏻

**Pull request merged**